projects/plugins/crm/includes/ZeroBSCRM.DAL3.php

```diff
@@ -571,6 +571,57 @@ public function fields_to_hide_on_frontend( $obj_type_int = false ){
 
     }
 
+	/**
+	 * Returns a count of objects of a type which are associated with a company or contact
+	 * Supports Quotes, Invoices, Transactions, Events currently
+	 *
+	 * @param int $cust_id The company / contact ID.
+	 * @param int $obj_type_id The type constant being checked (eg ZBS_TYPE_QUOTE).
+	 * @param int $zbs_type The contact or company type, for example ZBS_TYPE_COMPANY, ZBS_TYPE_CONTACT (default contact).
+	 *
+	 * @return int The count of relevant objects of the given type.
+	 */
+	public function customer_has_count_obj_type( $cust_id, $obj_type_id, $zbs_type = ZBS_TYPE_CONTACT ) {
```

Comment:
A nitpick and not a blocker, but this should probably be `contact_has_count_obj_type()`. If you do change it, make sure to update its references elsewhere.

Reply:
Now reading this again after looking through the code, I see that you've made this more generic. "Customer" is pretty consistently related to "Contact" in the CRM (and in large part we've tried to remove references to customer. Likewise, `has` typically means it'll return a `bool`. Perhaps instead:

`assigned_obj_type_count()`

I'm not in love with that function name either, so feel free to think on it!

Reply:
Yeah, I know we have history around the usage of customer - it did seem to be a compromise between contact and company so as not to be one or the other. How about `client_obj_type_count` using that same logic? 🤔

Reply:
"Client" is also used specifically in reference to "Contact" too (e.g. Client Portal). The function is generic, and would work with transactions assigned to invoices, for example, so I don't see a need to define it narrowly?

Reply:
> would work with transactions assigned to invoices,

I see! I honestly think your suggestion may be best - I've gone with `specific_obj_type_count_for_assignee` for now and modified the assignments, and related variables and comments.